applications/StructuralMechanicsApplication/custom_processes/set_automated_initial_variable_process.cpp

```diff
@@ -0,0 +1,124 @@
+      mThisParameters(ThisParameters)
+{
+    mThisParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void SetAutomatedInitialVariableProcess::ExecuteInitialize()
+{
+
+    KRATOS_TRY
+
+    array_1d<double, 3> hole_generatrix_axis = mThisParameters["hole_generatrix_axis"].GetVector();
+    KRATOS_ERROR_IF(MathUtils<double>::Norm3(hole_generatrix_axis) < std::numeric_limits<double>::epsilon()) << "The hole generatrix axis has norm zero" << std::endl;
+    
+    array_1d<double, 3> hole_generatrix_point = mThisParameters["hole_generatrix_point"].GetVector();  
+
+    array_1d<double, 3> normalized_generatrix_vector;   
+    normalized_generatrix_vector[0] = (hole_generatrix_axis[0] * hole_generatrix_axis[0]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+    normalized_generatrix_vector[1] = (hole_generatrix_axis[1] * hole_generatrix_axis[1]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+    normalized_generatrix_vector[2] = (hole_generatrix_axis[2] * hole_generatrix_axis[2]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+
+    const double hole_radius_offset = mThisParameters["hole_radius_offset"].GetDouble();
+    const double tol = 1.0e-6;
+
+    block_for_each(mrThisModelPart.Elements(), [&](Element &rElement) {
+
+        array_1d<double, 3> element_centroid = rElement.GetGeometry().Center();
+        
+        array_1d<double, 3> relative_position_vector;
+        relative_position_vector[0] = element_centroid[0] - hole_generatrix_point[0];
+        relative_position_vector[1] = element_centroid[1] - hole_generatrix_point[1];
+        relative_position_vector[2] = element_centroid[2] - hole_generatrix_point[2];
+
+        double vector_scaler = relative_position_vector[0] * normalized_generatrix_vector[0] + relative_position_vector[1] * normalized_generatrix_vector[1] + relative_position_vector[2] * normalized_generatrix_vector[2];
+        
+        array_1d<double, 3> intersection_point = hole_generatrix_point + vector_scaler * normalized_generatrix_vector;
+
+        array_1d<double, 3> radial_position_vector = element_centroid - intersection_point;
+
+        double centroid_relative_distance = std::sqrt(radial_position_vector[0] * radial_position_vector[0] + radial_position_vector[1] * radial_position_vector[1] + radial_position_vector[2] * radial_position_vector[2]) - hole_radius_offset;
+
+        if (centroid_relative_distance < 0.0 && abs(centroid_relative_distance) <= tol) {
```

Comment:
```suggestion
        if (centroid_relative_distance < 0.0 && std::abs(centroid_relative_distance) <= tol) {
```
Dangerous, abs is for integers and may return 0 with 0.21 FE